arc/conformance/plugin/arc_conformance.sh

```diff
@@ -0,0 +1,166 @@
+
+  kubectl config set-context ${KUBECTL_CONTEXT} \
+    --cluster=${KUBECTL_CONTEXT} \
+    --user=${KUBECTL_CONTEXT} \
+    --namespace=default 2> "${results_dir}"/error || python3 /arc/setup_failure_handler.py
+
+  kubectl config use-context ${KUBECTL_CONTEXT} 2> "${results_dir}"/error || python3 /arc/setup_failure_handler.py
+  echo "INFO: KubeConfig setup complete"
+}
+
+# validate enviorment variables
+if [ -z "${TENANT_ID}" ]; then
+  echo "ERROR: parameter TENANT_ID is required." > "${results_dir}"/error
+  python3 /arc/setup_failure_handler.py
+fi
+
+if [ -z "${SUBSCRIPTION_ID}" ]; then
+  echo "ERROR: parameter SUBSCRIPTION_ID is required." > "${results_dir}"/error
+  python3 /arc/setup_failure_handler.py
+fi
+
+if [ -z "${AZURE_CLIENT_ID}" ]; then
+  echo "ERROR: parameter AZURE_CLIENT_ID is required." > "${results_dir}"/error
+  python3 /arc/setup_failure_handler.py
+fi
+
+if [ -z "${AZURE_CLIENT_SECRET}" ]; then
+  echo "ERROR: parameter AZURE_CLIENT_SECRET is required." > "${results_dir}"/error
+  python3 /arc/setup_failure_handler.py
+fi
+
+if [ -z "${ARC_CLUSTER_NAME}" ]; then
+  echo "ERROR: parameter ARC_CLUSTER_NAME is required." > "${results_dir}"/error
+  python3 /arc/setup_failure_handler.py
+fi
+
+if [ -z "${ARC_CLUSTER_RG}" ]; then
+  echo "ERROR: parameter ARC_CLUSTER_RG is required." > "${results_dir}"/error
+  python3 /arc/setup_failure_handler.py
+fi
+
+# OBJECT_ID is an id of the Service Principal created in conformance test subscription.
+if [ -z "${OBJECT_ID}" ]; then
+  echo "ERROR: parameter OBJECT_ID is required." > "${results_dir}"/error
+  python3 /arc/setup_failure_handler.py
+fi
+
+# add az cli extensions 
+az extension add --name k8s-extension
+
+# login with service principal
+az login --service-principal \
+  -u "${AZURE_CLIENT_ID}" \
+  -p "${AZURE_CLIENT_SECRET}" \
+  --tenant "${TENANT_ID}" 2> "${results_dir}"/error || python3 /arc/setup_failure_handler.py
+
+az account set --subscription "${SUBSCRIPTION_ID}" 2> "${results_dir}"/error || python3 /arc/setup_failure_handler.py
+
+# set environment variables
+setEnviornmentVariables
+
+# setup Kubeconfig
+setupKubeConfig
+
+# Wait for resources in ARC agents to come up
+echo "INFO: Waiting for ConnectedCluster to come up"
+waitSuccessArc="$(waitForArc)"
+if [ "${waitSuccessArc}" == false ]; then
+    echo "helm release azure-arc failed" > "${results_dir}"/error
+    python3 /arc/setup_failure_handler.py
+    exit 1
+else
+    echo "INFO: ConnectedCluster is available"
+fi
+
+# register the KubernetesConfiguration resource provider
+az provider register --namespace Microsoft.KubernetesConfiguration
+
+echo "INFO: Creating extension"
+az k8s-extension create \
+      --name appconfigurationkubernetesprovider \
+      --extension-type Microsoft.AppConfiguration \
+      --cluster-name "${ARC_CLUSTER_NAME}" \
+      --resource-group "${ARC_CLUSTER_RG}" \
+      --cluster-type managedClusters \
```

Comment:
Could we parameterize this property, like this how the flux do https://dev.azure.com/ArcValidationProgram/ArcValidationProgram/_git/ConformanceTesting/commit/251064a0db6c69e0f971bb9a3773cd1aaf398b3f?refName=refs/heads/main&path=/aks/pipeline/scripts/sonobuoy-plugin-args/flux-managed.txt